src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1101,21 +1109,19 @@ protected override void Dispose(bool finalize)
             // workspace is going away. unregister this workspace from work coordinator
             StopSolutionCrawler();
 
-            // We should consider calling this here. It is commented out because Solution event tracking was 
-            // moved from VisualStudioProjectTracker, which is never Dispose()'d.  Rather than risk the 
-            // UnadviseSolutionEvents causing another issue (calling into dead COM objects, etc), we'll just 
+            // We should consider calling this here. It is commented out because Solution event tracking was
+            // moved from VisualStudioProjectTracker, which is never Dispose()'d.  Rather than risk the
+            // UnadviseSolutionEvents causing another issue (calling into dead COM objects, etc), we'll just
             // continue to skip it for now.
             // UnadviseSolutionEvents();
 
             base.Dispose(finalize);
         }
 
-        public void EnsureEditableDocuments(IEnumerable<DocumentId> documents)
+        public void EnsureEditableDocuments(params string[] fileNames)
         {
             var queryEdit = (IVsQueryEditQuerySave2)DeferredState.ServiceProvider.GetService(typeof(SVsQueryEditQuerySave));
 
-            var fileNames = documents.Select(GetFilePath).ToArray();
```

Comment:
why not keep as it was, but just filter out null paths?

Reply:
reverted